swift/WorkflowUI/Sources/Container/WorkflowView.swift

```diff
@@ -0,0 +1,203 @@
+        self.content = content
+        self.workflow = workflow
+    }
+    
+    public var body: some View {
+        IntermediateView(
+            workflow: workflow,
+            onOutput: onOutput,
+            content: content)
+    }
+    
+}
+
+@available(iOS 13.0, *)
+extension WorkflowView where T.Output == Never {
+    
+    /// Convenience initializer for workflows with no output.
+    public init(workflow: T, content: @escaping (T.Rendering) -> Content) {
+        self.init(workflow: workflow, onOutput: { _ in }, content: content)
+    }
+    
+}
+
+@available(iOS 13.0, *)
+extension WorkflowView where T.Rendering == Content {
+    
+    /// Convenience initializer for workflows whose rendering type conforms to `View`.
+    public init(workflow: T, onOutput: @escaping (T.Output) -> Void) {
+        self.init(workflow: workflow, onOutput: onOutput, content: { $0 })
+    }
+    
+}
+
+@available(iOS 13.0, *)
+extension WorkflowView where T.Output == Never, T.Rendering == Content {
+    
+    /// Convenience initializer for workflows with no output whose rendering type conforms to `View`.
+    public init(workflow: T) {
+        self.init(workflow: workflow, onOutput: { _ in }, content: { $0 })
+    }
+    
+}
+
+// We use a `UIViewController/UIViewControllerRepresentable` here to drop back to UIKit because it gives us a predictable
+// update mechanism via `updateUIViewController(_:context:)`. If we were to manage a `WorkflowHost` instance directly
+// within a SwiftUI view we would need to update the host with the updated workflow from our implementation of `body`.
+// Performing work within the body accessor is strongly discouraged, so we jump back into UIKit for a second here.
+@available(iOS 13.0, *)
+fileprivate struct IntermediateView<T: Workflow, Content: View>: UIViewControllerRepresentable {
+    
+    var workflow: T
+    var onOutput: (T.Output) -> Void
+    var content: (T.Rendering) -> Content
+    
+    func makeUIViewController(context: UIViewControllerRepresentableContext<IntermediateView<T, Content>>) -> WorkflowHostingViewController<T, Content> {
+        WorkflowHostingViewController(workflow: workflow, content: content)
+    }
+    
+    func updateUIViewController(_ uiViewController: WorkflowHostingViewController<T, Content>, context: UIViewControllerRepresentableContext<IntermediateView<T, Content>>) {
+        uiViewController.content = content
+        uiViewController.onOutput = onOutput
+        uiViewController.update(to: workflow)
+    }
+    
+}
+
+@available(iOS 13.0, *)
+fileprivate final class WorkflowHostingViewController<T: Workflow, Content: View>: UIViewController {
+    
+    private let workflowHost: WorkflowHost<T>
+    private let hostingController: UIHostingController<RootView<Content>>
+    private let rootViewProvider: RootViewProvider<Content>
+    
+    var content: (T.Rendering) -> Content
+    var onOutput: (T.Output) -> Void
+    
+    private let (lifetime, token) = Lifetime.make()
```

Comment:
I don't see anywhere this lifetime is being disposed - am I just missing it, or is this a potential leak if this container is removed from the view hierarchy?

Reply:
Disposal happens when it is destroyed after its reference count reaches zero – which happens when its owner (the view controller) is removed from the hierarchy.

Reply:
Cool! Must be nice having deterministic memory management 😅